Apps/Src/ActivateArray.c

```diff
@@ -0,0 +1,25 @@
+/* Copyright (c) 2020 UT Longhorn Racing Solar */
+
+#include "ActivateArray.h"
+
+void Task_ArrayConnection(void *p_arg) {
+
+    (void) p_arg;
+
+    OS_ERR err;
+    CPU_TS  ts;
+
+    // Wait until driver turned the key to the right place
+    OSSemPend(&ActivateArray_Sem4, 0, OS_OPT_PEND_BLOCKING, &ts, &err);
```

Comment:
What's setting this semaphore? Why do we need a separate task to enable the array, when the task that is reading from the switch to set this semaphore could instead just be calling an enableArray() or disableArray() function?

Reply:
I'm not quite sure what the original rationale was. As we have them right now, the activate array task and activate motor task turn on the precharge board, wait 5 seconds, and then turn on their respective contactor. Since the key will be turned quickly through both of these positions, we can't really have a blocking call to enableArray() (and a similar blocking call to enableMotor() ).

As for turning the array off, I think we can simply have some disableArray() function, since there's no need to wait for the precharge board. This function hasn't been written yet, however.

Reply:
From what I remember, to answer Chase's first question, the Switches reading task will set the semaphore once the key has been turned appropriately. The main rationale for a separate task, as Roie said, was to avoid blocking other functionality during the wait time. If that's not something that you think is necessary @chaseblock, it would be much simpler to put just a function call instead of a task. 

My opinion on this is that as long as this task isn't causing problems related to context switching (which I doubt it will since it'll be called very infrequently), I think it's fine in its current implementation. However, I would recommend adding the functionality to switch off the array with this as well. I recently changed the name of the task in the design to `ArrayConnection` mainly to imply that the task handles both directions of connection. I'll leave it up to you @Cam0Cow how exactly you want to implement it though.